cmd/cli/serve/flags.go

```diff
@@ -0,0 +1,262 @@
+package serve
+
+import (
+	"fmt"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+type flagDefinition struct {
+	Name        string
+	Path        string
+	Default     interface{}
+	Description string
+}
+
+// Generic function to define a flag, set a default value, and bind it to Viper
+func registerFlags(register map[string][]flagDefinition) error {
+	for name, defs := range register {
+		fset := pflag.NewFlagSet(name, pflag.ContinueOnError)
+		// Determine the type of the default value
+		for _, def := range defs {
+			switch v := def.Default.(type) {
+			case int:
+				fset.Int(def.Name, v, def.Description)
+			case bool:
+				fset.Bool(def.Name, v, def.Description)
+			case string:
+				fset.String(def.Name, v, def.Description)
+			case []string:
+				fset.StringSlice(def.Name, v, def.Description)
+			case map[string]string:
+				fset.StringToString(def.Name, v, def.Description)
+			}
+			viper.SetDefault(def.Path, def.Default)
+			if err := viper.BindPFlag(def.Path, fset.Lookup(def.Name)); err != nil {
+				return fmt.Errorf("")
+			}
```

Comment:
Binding the flag value to a string that also adds structure to the config.toml, i.e. `Node.Libp2p.SwarmPort` has a value of `1234`, `Node.Libp2p.PeerConnect` has value `none`. Viper will produce the below toml when writing a config:
```toml
[node.libp2p]
peerconnect = 'none'
swarmport = 1235
```
If the flag's value was bound to the flag name (e.g. `swarm-port`) we'd get a flat config, which I thought was a bit gross and hard to edit. These paths are also how we access values from viper, e.g. `viper.GetInt("Node.Libp2p.SwarmPort")`. This pattern allows us to have simple flags names that don't need to reflect config layout, and also allows users to generate a config file on demand based on their current bacalhau configuration from passed flags or env vars.